src/main/java/bio/terra/workspace/service/workspace/flight/CreateProjectStep.java

```diff
@@ -0,0 +1,151 @@
+
+  public CreateProjectStep(
+      CloudResourceManagerCow resourceManager,
+      ServiceUsageCow serviceUsage,
+      GoogleWorkspaceConfiguration googleWorkspaceConfiguration) {
+    this.resourceManager = resourceManager;
+    this.serviceUsage = serviceUsage;
+    this.googleWorkspaceConfiguration = googleWorkspaceConfiguration;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext) throws RetryException {
+    String projectId = flightContext.getWorkingMap().get(GOOGLE_PROJECT_ID, String.class);
+    createProject(projectId);
+    enableServices(projectId);
+    // TODO(PF-186): setup billing.
+    return StepResult.getStepResultSuccess();
+  }
+
+  private void createProject(String projectId) throws RetryException {
+    try {
+      Optional<Project> alreadyCreatedProject = retrieveProject(projectId);
+      if (alreadyCreatedProject.isPresent()) {
+        return;
+      }
+      Project project =
+          new Project()
+              .setProjectId(projectId)
+              .setParent(
+                  new ResourceId()
+                      .setType("folder")
+                      .setId(googleWorkspaceConfiguration.getFolderId()));
+      OperationCow<?> operation =
+          resourceManager
+              .operations()
+              .operationCow(resourceManager.projects().create(project).execute());
+      pollUntilSuccess(operation, Duration.ofSeconds(30), Duration.ofMinutes(5));
+    } catch (IOException e) {
+      throw new RetryException("Error creating project.", e);
+    }
+  }
+
+  private Optional<Project> retrieveProject(String projectId) throws IOException {
+    try {
+      return Optional.of(resourceManager.projects().get(projectId).execute());
+    } catch (GoogleJsonResponseException e) {
+      if (e.getStatusCode() == 403) {
+        // Google returns 403 for projects we don't have access to and projects that don't exist.
+        // We assume in this case that the project does not exist, not that somebody else has
+        // created a project with the same random id.
+        return Optional.empty();
+      }
+      throw e;
+    }
+  }
+
+  private void enableServices(String projectId) throws RetryException {
+    try {
+      Project project = resourceManager.projects().get(projectId).execute();
+      String projectName = "projects/" + projectId;
+      OperationCow<?> operation =
+          serviceUsage
+              .operations()
+              .operationCow(
+                  serviceUsage
+                      .services()
+                      .batchEnable(
+                          projectName,
+                          new BatchEnableServicesRequest().setServiceIds(ENABLED_SERVICES))
+                      .execute());
+      pollUntilSuccess(operation, Duration.ofSeconds(30), Duration.ofMinutes(5));
+    } catch (IOException e) {
+      throw new RetryException("Error enabling services.", e);
+    }
+  }
```

Comment:
I believe this is the operation we've seen take the longest and be the most unreliable. No change necessary for this PR, but keep in mind for the future ... it may be easier to debug/retry/etc if create-project and enable-services are separate stairway steps.